src/state_space/mod.rs

```diff
@@ -142,26 +160,52 @@ where
         StateSpaceBuilder { filters, ..self }
     }
 
-    pub async fn sync(self) -> Result<StateSpaceManager<T, N, P>, AMMError> {
-        let chain_tip = BlockId::from(self.provider.get_block_number().await?);
-        let factories = self.factories.clone();
+    pub fn with_checkpoint(self, path: impl AsRef<Path>) -> StateSpaceBuilder<T, N, P> {
+        StateSpaceBuilder {
+            checkpoint_path: Some(path.as_ref().canonicalize().unwrap()),
+            ..self
+        }
+    }
+
+    pub fn load_checkpoint(mut self) -> StateSpaceBuilder<T, N, P> {
```

Comment:
Lets make this `pub fn load_checkpoint(&mut self) -> SomeError`, then we can just call `self.load_checkpoint()` without having to bind to a variable.